dask_cuda/benchmarks/utils.py

```diff
@@ -344,45 +496,180 @@ def print_key_value(key, value, key_length=25):
     print(f"{key: <{key_length}} | {value}")
 
 
-def peer_to_peer_bandwidths(incoming_logs, scheduler_workers, ignore_size):
-    """Collect and aggregate peer-to-peer bandwidths"""
-    bandwidths = defaultdict(list)
-    total_nbytes = defaultdict(list)
-    bandwidths_all = []
-    for k, L in incoming_logs.items():
-        for d in L:
-            if d["total"] >= ignore_size:
-                bandwidths[k, d["who"]].append(d["bandwidth"])
-                total_nbytes[k, d["who"]].append(d["total"])
-                bandwidths_all.append(d["bandwidth"])
-    bandwidths = {
-        (scheduler_workers[w1].name, scheduler_workers[w2].name): [
-            "%s/s" % format_bytes(x) for x in np.quantile(v, [0.25, 0.50, 0.75])
-        ]
-        for (w1, w2), v in bandwidths.items()
-    }
-    total_nbytes = {
-        (
-            scheduler_workers[w1].name,
-            scheduler_workers[w2].name,
-        ): format_bytes(sum(nb))
-        for (w1, w2), nb in total_nbytes.items()
-    }
+def print_throughput_bandwidth(
+    args, durations, data_processed, p2p_bw, address_to_index
+):
+    print_separator(separator="=")
+    print_key_value(key="Wall clock", value="Throughput")
+    print_separator(separator="-")
+    durations = np.asarray(durations)
+    data_processed = np.asarray(data_processed)
+    throughputs = data_processed / durations
+    for duration, throughput in zip(durations, throughputs):
+        print_key_value(
+            key=f"{format_time(duration)}", value=f"{format_bytes(throughput)}/s"
+        )
+    print_separator(separator="=")
+    print_key_value(
+        key="Throughput",
+        value=f"{format_bytes(hmean(throughputs))}/s "
+        f"+/- {format_bytes(hstd(throughputs))}/s",
+    )
+    # Hack
+    bandwidth_hmean = p2p_bw[..., BandwidthStats._fields.index("hmean")].reshape(-1)
```

Comment:
```
np.asarray([tuple(1,2,3)], dtype=object) == np.asarray([[1,2,3]], dtype=object)
```
Since `BandwidthStats` is actually just a tuple, when the p2p_bw array is created we lose information about the `BandwidthStats` structure, so rather than having a 2D array of `BandwidthStats` objects, we have a 3D array where the last index can be indexed by the tuple indices of `BandwidthStats`.

So I'm accessing the internal `_fields` property of `BandwidthStats` to figure out which index I need for the mean (rather than hard-coding `0`).

It's probably fine

Reply:
But now that I look, `_fields` is part of the [public API](https://docs.python.org/dev/library/typing.html#typing.NamedTuple) so I am happy with this and will remove the hack comment.